### DISCUSSION TITLE
Livewire loads models anew!

### DISCUSSION THREAD

**Discussion by Terumi** [livewire/livewire#4242] [1. Help]
Hello people, 

I have a livewire component which takes a user list as a parameter and displays it in a table.
The problem is that the user model has a lot of relations and livewire loads everything for each model passed (even if I don't want my livewire component do anything with the extra data).
That gives me a lot of database queries which speed down my application.
Has anyone encountered this?
What I want is livewire not loading all these data anew.
Is this a normal behavior?
Are there any guidelines on passing a collection of model into a livewire component?

Thank you

**Comment by joshhanley** [comment]
@Terumi if you haven't eager loaded your relations before passing the list to the component, then I suggest doing that to stop n+1 problems.

But the way Livewire works is, every time a request is made, it needs to build up the state in your component from scratch (as that is how php works). To do this though, it needs to make database requests to get the models out of the database again and populate them into your component properties.

If you don't want this to happen, then you should not use eloquent models/ eloquent collections in your properties, instead you should convert your models to arrays.

Hope this helps!
